src/sage/groups/finitely_presented.py

```diff
@@ -804,6 +808,10 @@ def __init__(self, free_group, relations, category=None):
         self._free_group = free_group
         self._relations = relations
         self._assign_names(free_group.variable_names())
+        if category is None:
+            category = Groups().FinitelyGenerated()
+        else:
+            category &= Groups().FinitelyGenerated()
         parent_gap = free_group.gap() / libgap([rel.gap() for rel in relations])
```

Comment:
I agree with `if` part, but if the category is given by the user, shouldn't we use it as given, instead of joining with `Groups().FinitelyGenerated()`? Do you have a reason for doing that?

Reply:
well, this is adding information, so there is no harm and this could be useful

Reply:
As far as I know, the idiom is just to pass the user-supplied category on.

> this is adding information, ...

I imagine that someone might not want the additional information...

But really I don't have a substantial argument. So OK if you insist.